Framework/Muon/src/ConvertFitFunctionForMuonTFAsymmetry.cpp

```diff
@@ -270,8 +270,10 @@ std::vector<double> ConvertFitFunctionForMuonTFAsymmetry::getNorms() {
 
   std::vector<double> norms(wsNames.size(), 0);
   auto colNames = table->getColumnNames();
-  auto wsNamesIndex = findName(colNames, "name");
-  auto normIndex = findName(colNames, "norm");
+  std::string colName = "name";
```

Comment:
@AnthonyLim23 I think the problem here is that `findName` has the `name` variable taken by reference and not by value. If it is taken by value then you won't need these temporary variables (basically remove the `&` from `std::string &name`).

Reply:
but when I did that the cpp check was unstable

Reply:
Ah I see. I guess in this case it actually makes little difference as you have to construct a string temporary. Maybe a simpler way, given the usage, is to change to `int findName(const std::vector<std::string>& colNames, const char* name)` and then we avoid the string allocation entirely.